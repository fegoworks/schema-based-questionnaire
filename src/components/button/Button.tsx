import "./button.scss";
import { FunctionComponent } from "react";
import { Link } from "react-router-dom";

interface ButtonArgs {
  isLoading?: boolean;
  onClick?: (e: any) => void;
  text: string;
  to?: string;
  type: "button" | "link" | "submit" | "reset";
  disabled?: boolean;
}
const Button: FunctionComponent<ButtonArgs> = ({
  isLoading,
  onClick,
  text,
  to = "#",
  type = "button",
  disabled,
}) => {
  return type === "link" ? (
    <a onClick={onClick} href={to} className="link-button">
      {text}
    </a>
  ) : (
    <button
      disabled={isLoading || disabled}
      type={type}
      onClick={onClick && onClick}>
      {isLoading ? "Loading..." : text}
    </button>
  );
};

export default Button;
