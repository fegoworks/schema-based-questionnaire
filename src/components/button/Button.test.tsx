import { fireEvent, render } from "@testing-library/react";
import { describe, expect, vi } from "vitest";
import Button from "./Button";

describe("Component - Button", () => {
  let onClick: () => void;

  beforeEach(() => {
    onClick = vi.fn();
  });

  it("should render button component", () => {
    const { container } = render(<Button text="Click Me!" type="button" />);
    expect(container).toMatchSnapshot();
  });

  it("should render the link button", () => {
    const { getByRole } = render(<Button text="Click Me!" type="link" />);
    expect(getByRole("link")).toHaveClass("link-button");
  });

  it("should fire the onClick function when clicked for active button", () => {
    const { getByRole } = render(
      <Button onClick={onClick} text="Click me" type="button" />
    );

    fireEvent.click(getByRole("button"));
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it("should not fire the onClick function for disabled buttons", () => {
    const { getByRole } = render(
      <Button onClick={onClick} text="Click Me!" type="button" disabled />
    );
    fireEvent.click(getByRole("button"));
    expect(onClick).toHaveBeenCalledTimes(0);
  });
});
