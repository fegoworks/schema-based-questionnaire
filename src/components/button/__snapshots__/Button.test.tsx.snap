// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component - Button > should render button component 1`] = `
<div>
  <button
    type="button"
  >
    Click Me!
  </button>
</div>
`;
